docs/cache.txt

```diff
@@ -0,0 +1,240 @@
+driver, but recommended, because they provide better performance by delegating the
+deletion of expired documents to MongoDB instead of requiring the application to
+perform this task randomly.
+
+Create the indexes with a migration that calls the
+``createTTLIndex()`` methods provided by both the cache, and the lock stores:
+
+.. literalinclude:: /includes/cache/migration.php
+   :language: php
+   :emphasize-lines: 11,12
+   :dedent:
+
+Then run the migration:
+
+.. code-block:: none
+
+   php artisan migrate
+
+Alternatively, you can create the index by using :mdb-shell:`MongoDB Shell <>` (``mongosh``):
+
+.. code-block:: ts
+
+   db.cache.createIndex(
+     /* Field that holds the expiration date */
+     { expires_at: 1 },
+     /* Delay to remove items after expiration */
+     { expireAfterSeconds: 0 }
+   )
+
+If you use Locks, disable ``lock_lottery`` by setting the probability to ``0``:
+
+.. code-block:: php
+   :highlightLines: [4]
+
+   'stores' => [
+       'mongodb' => [
+           'driver' => 'mongodb',
+           'connection' => 'mongodb',
+           'lock_lottery' => [0, 100], // Disabled
+       ],
+   ],
+
+Using MongoDB Cache
+-------------------
+
+The Laravel cache can be used to store any serializable data using the facade
+``Illuminate\Support\Facades\Cache``.
+
+This example performs the following actions:
+
+- Gets the cache repository with the ``mongodb`` store
+- Tries to read and return the cache item named ``foo``
+- If missing, calls the closure to compute the value, stores the value forever, and returns it
+
+.. code-block:: php
+
+   use Illuminate\Support\Facades\Cache;
+
+   $value = Cache::store('mongodb')->get('foo', function () {
+       return [1, 2, 3];
+   });
+
+By default, cached objects do not expire. However, it is possible to define an expiry time, as shown in the following example:
+
+.. code-block:: php
+
+   Cache::store('mongodb')->set('foo', 'abc', '1 day');
+
+Incrementing and decrementing a value is also supported if the value is
+initialized before. The following example initializes the counter to ``3``,
+adds 5, and removes 2.
+
+.. code-block:: php
+
+   Cache::store('mongodb')->set('counter', 3);
+   Cache::store('mongodb')->increment('counter', 5);
+   Cache::store('mongodb')->decrement('counter', 2);
+
+.. note::
+
+   {+odm-short+} supports incrementing and decrementing with integer and float values.
+
+For more information on using cache, see the `Laravel Cache documentation
+<https://laravel.com/docs/{+laravel-docs-version+}/cache#cache-usage>`__.
+
+Configuring MongoDB as Default Cache
+------------------------------------
+
+To use the ``mongodb`` store by default, change the default store in
+``config/cache.php``:
+
+.. code-block:: php
+   :emphasize-lines: 2
+
+   return [
+       'default' => env('CACHE_STORE', 'mongodb'),
+
+       'stores' => [
+           'mongodb' => [
+               'driver' => 'mongodb',
+               'connection' => 'mongodb',
```

Comment:
Should `collection` and various `lock` options be specified here?

Also, is there any significance to the line break between `default` and `stores` in the configuration array? I believe you're already emphasizing the `default` line, so I don't think a line break is necessary for readability.

Reply:
Here, I rely on the default values. Do you think I should make a note of that?

Reply:
Calling out your reliance on default values makes sense to me.

I was mostly concerned with `collection`. The MongoLock constructor requires a value for its Collection parameter, and I didn't see where the default gets supplied.

Reply:
"co**nn**ection" is required, "co**ll**ection" has a default value which is `cache`. These 2 words are so similar that they can easily be confused.